lib/Text/Glob.pm

```diff
@@ -177,6 +189,10 @@ the regex, or change the seperator that Text::Glob uses by setting
 C<$Text::Glob::seperator> to an alternative value while compiling the
 the regex.
 
+However, if C<$Text::Glob::globstar> is set to a true value, two
+consecutive asterisks (e.g. C<**.foo>) B<will> match C</>, though one
+asterisk will not.
+
```

Comment:
Hello, I recently stumbled across this because I wanted to match file names like rsync does.  (I had not heard about the feature being in bash before)

I was going to work out the patch that would allow ``foo/**/bar`` to match ``foo/bar``, but it turned out to be a lot more complicated than I expected.  First, bash doesn't actually allow `**` to match a `/`, it allows it to match zero or more directories.  If you create a file ``a/b/c.pm``,  ``echo a**.pm`` does not match it.  It seems that any time double-star is not anchored to a directory separator or the end of the string, it acts like a normal single-star.  Also, bash does not allow double-star to match hidden directories, so the implementation should probably respect ``$strict_leading_dot`` in its implementation of ``$globstar``.

Meanwhile, I actually like the rsync version of ``**`` better, which does match partial filenames at the same time as directories.  (but it's not a big deal, since rsync ``**`` can be represented with bash ``*/**/*`` and get the same effect)  So maybe the first question is whether we want something compatible with bash, or compatible with rsync?

Also I found a bug where ``{_,.}foo`` doesn't match ``.foo`` when strict-dot is in effect, which I was going to fix with the implementation of strict-dot-aware globstar.